docs/shone/examples/opacities.rst

```diff
@@ -0,0 +1,267 @@
+
+Let's load the opacity created by `~shone.opacity.generate_synthetic_opacity()`
+(see the step above):
+
+.. code-block:: python
+
+    opacity = Opacity.load_species_from_name('synthetic')
+
+The `~shone.opacity.Opacity` object contains the opacity grid as a `~xarray.DataArray`
+in its `grid` attribute. You can see the dimensions of the grid with:
+
+.. code-block:: python
+
+    >>> print(opacity.grid.coords)
+    Coordinates:
+      * wavelength   (wavelength) float64 0.5 0.5012 0.5023 ... 4.977 4.988 5.0
+      * temperature  (temperature) int32 200 400 600 800 1000
+      * pressure     (pressure) float64 1e-06 10.0
+
+The coordinates in the `~xarray.DataArray` are wavelength in microns,
+temperature in K, and pressure in bar. To learn to use the xarray API
+directly on the grid attribute, refer to the xarray docs on `indexing
+and selecting data <https://docs.xarray.dev/en/stable/user-guide/indexing.html>`_
+and `interpolating
+<https://docs.xarray.dev/en/stable/user-guide/interpolation.html>`_.
+
+You can inspect the opacities from one temperature and pressure slice like so:
+
+.. code-block:: python
+
+    import matplotlib.pyplot as plt
+
+    opacity_sample = opacity.grid.sel(
+        dict(
+            pressure=10,  # [bar]
+            temperature=200  # [K]
+        )
+    )
+
+    plt.semilogy(
+        opacity_sample.wavelength, opacity_sample.opacity,
+        label=f"T={opacity_sample.temperature} K"
+    )
+    plt.gca().set(
+        xlabel='Wavelength [µm]',
+        ylabel='Opacity, $\kappa$ [cm$^2$ g$^{-1}$]'
+    )
+
+.. plot::
+
+    import matplotlib.pyplot as plt
+    from shone.opacity import Opacity, generate_synthetic_opacity
+
+    opacity = generate_synthetic_opacity()
+    opacity_sample = opacity.grid.sel(
+        dict(
+            pressure=10,  # [bar]
+            temperature=200  # [K]
+        )
+    )
+
+    plt.semilogy(
+        opacity_sample.wavelength, opacity_sample,
+        label=f"T={opacity_sample.temperature} K"
+    )
+    plt.gca().set(
+        xlabel='Wavelength [µm]',
+        ylabel='Opacity, $\kappa$ [cm$^2$ g$^{-1}$]'
+    )
+
+Interpolating opacities
+-----------------------
+
+Often in ``shone`` we will need to interpolate over the opacity grid within
+compiled code, so we will use a just-in-time compiled interpolator on the
+opacity grid. You can produce a function to do these compiled interpolations
+with `~shone.opacity.Opacity.get_interpolator`:
+
+.. code-block:: python
+
+    # get a jitted 3D interpolator over wavelength, temperature, pressure:
+    interp_opacity = opacity.get_interpolator()
+
+Now you can get the opacity at wavelengths, temperatures, and pressures that weren't on
+the grid:
+
+.. code-block:: python
+
+    wavelength = np.linspace(1, 5, 500)  # [µm]
+    pressure = 0.3  # [bar]
+    temperature = 555  # [K]
+
+    example_opacity = interp_opacity(wavelength, temperature, pressure)
+
+    plt.semilogy(wavelength, example_opacity, label=f"T={temperature} K")
```

Comment:
I suggest to overplot the opacity functions at the temperatures of the grid points to compare. If the synthetic opacity is T-dependent ofc.

Reply:
Wouldn't you say that's sort of redundant? The point of H&K 2017 is that R ~ ln(kappa).

Reply:
No this was mostly just for visual purposes. To show that the interpolated opacity lands between two grid entries. Not necessary though.